task-forage.lic

```diff
@@ -0,0 +1,621 @@
+      2545   => ['e'],
+      2621   => ['e', 'w', 'w', 'w', 'w', 'w', 'go wrens', 'nw', 'sw'],
+      2613   => ['ne'],
+      2608   => ['ne'],
+      2617   => ['ne'],
+      2633   => ['ne'],
+      2632   => ['ne'],
+      2631   => %w[sw sw sw sw sw sw nw],
+      2504   => ['nw'],
+      2505   => ['nw'],
+      2506   => ['nw'],
+      2507   => ['nw'],
+      2508   => ['nw'],
+      2509   => ['go brid', 'ne', 'e', 'e', 'e', 'e', 'se'],
+      2574   => ['nw'],
+      2540   => ['w'],
+      2539   => %w[e se se],
+      2541   => ['se'],
+      2543   => ['nw', 'nw', 'go brid', 'sw', 'sw', 'sw', 'sw', 'sw', 'sw', 'e'],
+      2603   => ['se'],
+      2602   => %w[s s sw w sw],
+      2583   => ['sw'],
+      2584   => ['sw'],
+      2592   => ['sw'],
+      2593   => ['sw'],
+      2594   => ['sw'],
+      2595   => ['go brid', 'se', 'e', 'e', 'n', 'n', 'n', 'n', 'n', 'w'],
+      2571   => ['sw'],
+      2572   => ['s'],
+      2569   => ['se'],
+      2568   => %w[e s s s s e e ne ne ne n]
+    }
+
+    @move_regex = [/The flower peddler gathers her skirts and glides (<?direction>north|south|east|west|northeast|southeast|southwest|northwest)\./]
+    Flags.add('combat', 'You cannot forage while in combat!', 'closes to pole weapon range on you!',
+              'begins to advance on you!')
+    Flags.add('giver-move', *@move_regex)
+
+    @task_givers = { 'shard'    => { 'npc' => 'peddler', 'location' => @shard_path },
+                     'crossing' => { 'npc' => 'Mags', 'location' => '954' } }
+
+    UserVars.task_forage = {} if UserVars.task_forage.nil?
+    # Track how the last time you were unable to forage an item
+    UserVars.task_forage['item_failures'] = {} if UserVars.task_forage['item_failures'].nil?
+    # Used to store the last known location of wandering task givers
+    UserVars.task_forage['npcs'] = {} if UserVars.task_forage['npcs'].nil?
+
+    @task_town = (@args.town.nil? ? @hometown : @args.town).downcase
+    @task_giver = @task_givers[@task_town]['npc']
+
+    @forage_container = @forage_settings['container'] || @settings.forage_container || 'backpack'
+    @trading_limit = @forage_settings['trading_limit'] || 30
+    @never_wait = @forage_settings['never_wait'].nil? ? false : @forage_settings['never_wait']
+    @use_boosts = @forage_settings['task_boosts'].nil? ? false : @forage_settings['task_boosts']
+    @collect = @forage_settings['collect'].nil? ? false : @forage_settings['collect']
+    @debug = @forage_settings['debug'].nil? ? false : @forage_settings['debug']
+
+    @boosted = nil
+    @number = nil
+    @item = nil
+    @item_location = nil
+    @tasks_completed = 0
+    @tasks_failed = 0
+
+    echo 'UserVars:' if @debug
+    echo UserVars.task_forage.to_yaml if @debug
+    echo 'forage_settings:' if @debug
+    echo @forage_settings.to_yaml if @debug
+
+    while determine_task_necessary
+      find_giver
+      until get_task
+        # loop until we have an appropriate foraging task
```

Comment:
Would it not make more sense to make get_task method a loop internally? replace false with next and true with break.